### PR TITLE
Add release notes for v20.2.2

### LINKED ISSUE #8919: v20.2.2 Release notes
Prep date: 11/16/20
Release date: 11/23/20
SHA: `36822fee607976556f265c683da0eaffcfdfec8b`
Release Manager: Nathan
Issue: https://github.com/cockroachdb/cockroach/issues/56757

[Docs To Do](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-16.Updatedocs):

- [ ] Run script to generate release notes
- [ ] Update docs site
- [ ] Edit release notes - Additions/changes/omissions to the release notes should be left in this issue's comments
- [ ] Once PR is merged / release notes are live, notify @meagan-g in the #releases Slack channel
- [ ] Email releases@ with link to published release notes
- [ ] Generate release notes issues (cc @lnhsingh)

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes #8919.

### CODE REVIEW

**Review by lnhsingh — COMMENTED** [review]
Once nits are fixed, LGTM!

Also just a reminder to update the release date where necessary if this doesn't go out on Monday.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8965#-:-MMXBj5ADpnvt4noXMSp:bqaut3z)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @lnhsingh and @rmloveland)

___
*[_data/releases.yml, line 3 at r1](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMX1QlG5CfbTuJ8LUg4:-MMX1QlG5CfbTuJ8LUg5:bk8jmz4) ([raw file](https://github.com/cockroachdb/docs/blob/488821272f132c92bd7371bbd8e85d3c5fe95f7e/_data/releases.yml#L3)):*
> ```YAML
> - title: Production releases
>   releases:
>     - date: Nov 10, 2020
> ```

reminder to update this date to the new release date
___
*[releases/v20.2.2.md, line 47 at r1](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMX82xD6EKLC-5xdBPX:-MMX82xE9eoIq4Rr28kG:bp2ospx) ([raw file](https://github.com/cockroachdb/docs/blob/488821272f132c92bd7371bbd8e85d3c5fe95f7e/releases/v20.2.2.md#L47)):*
> ```Markdown
> ### Enterprise edition changes
> 
> - The `insecure_tls_skip_verify` query string parameter may now be set on [changefeed](../v20.2/stream-data-out-of-cockroachdb-using-changefeeds.html) sinks. This disables client-side validation of responses and should be avoided if possible since it creates man-in-the-middle vulnerabilities unless combined with another method of authentication. [#56338][#56338]
> ```

nit: from a quick google, it seems like "man-in-the-middle" is a known term, but could we substitute it for a more inclusive one? Wikipedia offers other options like "machine-in-the-middle" or "person-in-the-middle"

https://en.wikipedia.org/wiki/Man-in-the-middle_attack
___
*[releases/v20.2.2.md, line 54 at r1](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMX8WbQ79BFj12texSr:-MMX8WbQ79BFj12texSs:b-drxmkb) ([raw file](https://github.com/cockroachdb/docs/blob/488821272f132c92bd7371bbd8e85d3c5fe95f7e/releases/v20.2.2.md#L54)):*
> ```Markdown
> - Parsing [intervals](../v20.2/interval.html) with fractional years now produces intervals with no more precision than months, to match the behavior of Postgres. [#56158][#56158]
> - The `pg_attribute.atttypmod` column in the `pg_catalog` is now populated for [collated string types](../v20.2/collate.html). This also populates the value of the `TypeModifier` in the `RowDescription` message of the [pgwire protocol](https://www.postgresql.org/docs/current/static/protocol.html). [#55154][#55154]
> - Add an option to [scheduled backups](../v20.2/manage-a-backup-schedule.html) to maintain a timeseries metric for last backed up timestamp. [#54987][#54987]
> ```

nit: Add > Added
___
*[releases/v20.2.2.md, line 59 at r1](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMX8aHM-UiuY8ibdDyB:-MMX8aHM-UiuY8ibdDyC:blff6lk) ([raw file](https://github.com/cockroachdb/docs/blob/488821272f132c92bd7371bbd8e85d3c5fe95f7e/releases/v20.2.2.md#L59)):*
> ```Markdown
> ### API endpoint changes
> 
> - `AWS_SERVER_ENC_MODE` and `AWS_SERVER_KMS_ID` can now be specified as parameters in all s3 URIs which write to the store. This change affects both [changefeeds](../v20.2/stream-data-out-of-cockroachdb-using-changefeeds.html) and [`BACKUP`](../v20.2/backup.html)s. [#56132][#56132]
> ```

nit: s3 > S3, which > that
___
*[releases/v20.2.2.md, line 65 at r1](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMX9Xoq4ByA-CUpbmYE:-MMX9Xoq4ByA-CUpbmYF:b-1b7evi) ([raw file](https://github.com/cockroachdb/docs/blob/488821272f132c92bd7371bbd8e85d3c5fe95f7e/releases/v20.2.2.md#L65)):*
> ```Markdown
> - Fixed a bug where CockroachDB would crash when executing a query via the [vectorized engine](../v20.2/vectorized-execution.html) when most of the SQL memory (determined via `--max-sql-memory` argument to [`cockroach start`](../v20.2/cockroach-start.html)) had already been reserved. [#55457][#55457]
> - Fixed a very rare bug which could lead to possible write skew in distributed queries that have both zigzag joins and table readers with the zigzag joins reading keys not read by the table readers. [#55563][#55563]
> - The current implementation of [changefeeds](../v20.2/stream-data-out-of-cockroachdb-using-changefeeds.html) is incompatible with the [vectorized engine](../v20.2/vectorized-execution.html). Therefore, whenever the vectorized engine was being used to run changefeeds, the command could hang indefinitely. This bug is now fixed. On 20.2 releases this could happen if the user ran `SET vectorize_row_count_threshold=0;`, and on 20.1 releases it could happen if the user ran `SET vectorize=on`. [#55753][#55753]
> ```

nit: 20.2 / 20.1 > v20.2 / v20.1
___
*[releases/v20.2.2.md, line 67 at r1](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMX9a6bCuOe2xarK5CN:-MMX9a6bCuOe2xarK5CO:b-xzfvls) ([raw file](https://github.com/cockroachdb/docs/blob/488821272f132c92bd7371bbd8e85d3c5fe95f7e/releases/v20.2.2.md#L67)):*
> ```Markdown
> - The current implementation of [changefeeds](../v20.2/stream-data-out-of-cockroachdb-using-changefeeds.html) is incompatible with the [vectorized engine](../v20.2/vectorized-execution.html). Therefore, whenever the vectorized engine was being used to run changefeeds, the command could hang indefinitely. This bug is now fixed. On 20.2 releases this could happen if the user ran `SET vectorize_row_count_threshold=0;`, and on 20.1 releases it could happen if the user ran `SET vectorize=on`. [#55753][#55753]
> - Fixed a bug where CockroachDB could incorrectly evaluate the `sqrdiff` function when used as a [window function](../v20.2/window-functions.html) in some cases. [#55995][#55995]
> - Fixed a bug where CockroachDB could incorrectly compute some [aggregate functions](../v20.2/functions-and-operators.html#aggregate-functions) with `DISTINCT` clauses when the query projected other columns/functions and the [vectorized engine](../v20.2/vectorized-execution.html) was used. This bug was introduced in 20.2.0.alpha.3 release. [#55872][#55872]
> ```

nit: 20.2.0.alpha.3 release > the v20.2.0-alpha.3 release
___
*[releases/v20.2.2.md, line 81 at r1](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMXAoqz8WfxVQtAajgh:-MMXAoqz8WfxVQtAajgi:bpbw26d) ([raw file](https://github.com/cockroachdb/docs/blob/488821272f132c92bd7371bbd8e85d3c5fe95f7e/releases/v20.2.2.md#L81)):*
> ```Markdown
> - Fixed a bug where if a [cluster backup](../v20.2/backup.html#backup-a-cluster) was taken during a [schema change](../v20.2/online-schema-changes.html), a cluster restore of that backup would create duplicates of the ongoing schema changes. [#56390][#56390]
> - Fixed a bug where [dumps](../v20.2/cockroach-dump.html) of tables with a [`BIT`](../v20.2/bit.html) type column would result in an error. This column type is now supported. [#56391][#56391]
> - Fixed a bug which would cause [transactions](../v20.2/transactions.html) which modified [roles](../v20.2/authorization.html#create-and-manage-roles) and then attempted to read or modify other roles would encounter blocking and stale data. [#55392][#55392]
> ```

nit: would encounter > to encounter?

I also think the `which`s should be `that` but now I'm not sure
___
*[releases/v20.2.2.md, line 83 at r1](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMXBLdb8QqmMzGFerOm:-MMXBLdb8QqmMzGFerOn:b4cf0us) ([raw file](https://github.com/cockroachdb/docs/blob/488821272f132c92bd7371bbd8e85d3c5fe95f7e/releases/v20.2.2.md#L83)):*
> ```Markdown
> - Fixed a bug which would cause [transactions](../v20.2/transactions.html) which modified [roles](../v20.2/authorization.html#create-and-manage-roles) and then attempted to read or modify other roles would encounter blocking and stale data. [#55392][#55392]
> - Fixed a bug where CockroachDB did not respect disabling [protected timestamp settings `kv.protectedts.max_bytes` and `kv.protectedts.max_spans`](../v20.2/cluster-settings.html) by setting them to zero values. [#56453][#56453]
> - Fixed a panic that could occur when running [`SHOW STATISTICS USING JSON`](../v20.2/show-statistics.html) for a table in which at least one of the columns contained all _NULL_ values. [#56516][#56516]
> ```

Does NULL need to be italicized?
___
*[releases/v20.2.2.md, line 91 at r1](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMXBR4uCvBaiVqmHk8U:-MMXBR4uCvBaiVqmHk8V:bcytuhw) ([raw file](https://github.com/cockroachdb/docs/blob/488821272f132c92bd7371bbd8e85d3c5fe95f7e/releases/v20.2.2.md#L91)):*
> ```Markdown
> - Adjusted the cost model in the [optimizer](../v20.2/cost-based-optimizer.html) so that the optimizer is less likely to plan a [lookup join](../v20.2/joins.html#lookup-joins) into a virtual table. Performing a lookup join into a virtual table is expensive, so this change will generally result in better performance for queries involving joins with virtual tables. [#56349][#56349]
> 
> ### Miscellaneous
> ```

I think you can get rid of this heading and maybe move the `DB Console changes` to be above `Bug fixes`

**Review by lucy-zhang — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8965)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rmloveland)

___
*[releases/v20.2.2.md, line 129 at r2](https://reviewable.io/reviews/cockroachdb/docs/8965#-MMwQxi7Dn6UkiaCGyt3:-MMwQxi7Dn6UkiaCGyt4:bb6ywmx) ([raw file](https://github.com/cockroachdb/docs/blob/4a7ad84fd328a03b921f90495c106bef27dda384/releases/v20.2.2.md#L129)):*
> ```Markdown
> ### Bug fixes
> 
> - Fixed a bug introduced in v20.2 where we failed to upgrade [foreign keys](../v20.2/foreign-key.html) that used the pre-19.2-style internal representation when validating cross-references for tables.  This bug caused validation failures that made the referenced tables and metadata unavailable. The fix gets rid of the validation errors by accounting for the pre-19.2-style internal representation of foreign keys. [#57083][#57083]
> ```

LGTM

File: _data/releases.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 - title: Production releases
   releases:
+    - date: Nov 25, 2020
+      version: v20.2.2
+      latest: true
     - date: Nov 20, 2020
       version: v20.2.1
-      latest: true
-    - date: Nov 10, 2020
```
Comment (jseldess) [inline _data/releases.yml:6]: @rmloveland, this remove the 20.2.0 release notes, which has the full summary of all features in the major release, from the list of release, so there's currently no way to find the page unless you know it exists. Was this intentional? In any case, I'll open a PR to put it back.

Comment (rmloveland) [reply _data/releases.yml:6]: Definitely human error!

**Pull request merged**